ntp-proto/src/nts_record.rs

```diff
@@ -1288,7 +1288,13 @@ impl KeyExchangeServerDecoder {
                 Continue(state)
             }
 
-            Unknown { .. } => Continue(state),
+            Unknown { critical, .. } => {
+                if critical {
+                    Break(Err(KeyExchangeError::UnrecognizedCriticalRecord))
+                } else {
+                    Continue(state)
+                }
```

Comment:
original cause of this PR. we test now also that an unknown record with a critical bit sends a record to that effect to the client.